x-pack/platform/plugins/shared/actions/server/lib/single_file_connectors/generate_executor_function.ts

```diff
@@ -41,6 +41,7 @@ export const generateExecutorFunction = ({
       actionId: connectorId,
       config,
       connectorTokenClient,
+      globalAuthHeaders,
```

Comment:
Don't we have the headers when we call the wrapping function `generateExecutorFunction`?

This could be removed from `execOptions` and come from the wrapper that already has access to the headers from the spec.

Reply:
Wouldn't we just have to change [this line](https://github.com/elastic/kibana/blob/main/x-pack/platform/plugins/shared/actions/server/lib/single_file_connectors/create_connector_from_spec.ts#L27) to

```
  const executor = generateExecutorFunction({
    globalAuthHeaders: spec.auth?.headers.
    actions: spec.actions,
    getAxiosInstanceWithAuth: actions.getAxiosInstanceWithAuth,
  });
```

If it is hardcoded in the spec it can be used to generate the function.

Maybe I am interpreting this wrong 😅